src/main/java/uk/gov/pay/directdebit/mandate/services/MandateService.java

```diff
@@ -84,17 +87,13 @@ public Mandate createMandate(CreateMandateRequest createRequest, String accountE
         return gatewayAccountDao.findByExternalId(accountExternalId)
                 .map(gatewayAccount -> {
                     if (gatewayAccount.getAccessToken().isEmpty() &&
-                            gatewayAccount.getPaymentProvider() == PaymentProvider.GOCARDLESS) {
+                            gatewayAccount.getPaymentProvider() == GOCARDLESS) {
                         LOGGER.error("Gateway account with id {} does not have access token", accountExternalId);
                         throw new UnlinkedGCMerchantAccountException(accountExternalId);
                     }
 
-                    // TODO:
-                    // when we introduce GoCardless gateway accounts to work with create mandate,
-                    // then modify appropriate mandate reference values
-                    MandateBankStatementReference mandateReference = MandateBankStatementReference.valueOf(
-                            PaymentProvider.SANDBOX.equals(gatewayAccount.getPaymentProvider()) ?
-                                    RandomStringUtils.randomAlphanumeric(18) : "gocardless-default");
+                    var mandateReference = SANDBOX.equals(gatewayAccount.getPaymentProvider()) ?
+                            MandateBankStatementReference.valueOf(randomAlphanumeric(18)) : null;
```

Comment:
Shouldn't this be set when the mandate is confirmed for a consistent experience between Go Cardless and Sandbox accounts?